File: API/Models/Room.cs
namespace API.Models;

public class Room
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public double BasePrice { get; set; }
    public double AdditionalGuestPrice { get; set; }
    public int MaxGuests { get; set; }
    public int NumBeds { get; set; }
    public string PrimaryImageUrl { get; set; } = string.Empty;
    public List<string> SecondaryImageUrls { get; set; } = new List<string>();
    public List<Feature> Features { get; set; } = new List<Feature>();
}

File: client/src/interfaces/models/Room.ts
import { Feature } from "./Feature.ts";


export interface Room {
    id: number;
    name: string;
    basePrice: number;
    additionalGuestPrice: number;
    maxGuests: number;
    numBeds: number;
    primaryImageUrl: string;
    secondaryImageUrls: string[];
    features: Feature[];
}